components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -51,22 +61,41 @@ export class ProjectsFilterDropdownComponent implements OnChanges {
     }
   }
 
-  handleEscape() {
-    this.dropdownActive = false;
+  handleEscape(): void {
     this.resetOptions();
+    this.filterValue = '';
+    this.dropdownActive = false;
+    this.onOptionChange.emit(this.editableOptions);
   }
 
-  handleOptionChange(event, index) {
-    this.editableOptions[index].checked = event.detail;
+  handleOptionChange(event, label) {
+    this.editableOptions.find(option => {
+      if (option.label === label) {
+        option.checked = event.detail;
+      }
+    });
```

Comment:
I had to change this from using the index to using something else (i went with label), because the index is no longer correct when the array of options is being filtered.